### PR TITLE
beat that project

### PR DESCRIPTION
**Pull request by valexischang**
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment?
>24 hours
Please fill in one error and/or error message you received while working on this assignment.
undefined

What part of the assignment did you spend the most time on?
refactoring the code to account for multiple players
Comfort Level (1-5):
2
Completeness Level (1-5):
3
What did you think of this deliverable?
While I didn't manage to complete everything, I solidified my understanding of how to parse arguments into the function and learned how to make the code more dynamic.

Is there anything in this code that you feel pleased about?
A better understanding of how functions and loops work. Also forced to learn how to use console.log function more frequently to debug.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -1,4 +1,208 @@
...
+var userName = "";
+var totalPlayers = 0;
+var playerList = [];
+
+var numOfDiceSelected = function (num) {
+  var numOfDice = num;
+  if (num != Number(num)) {
+    return -1;
+  } else {
+    return Number(numOfDice);
```
Comment (emily-zhangg) [inline script.js:28]: an alternate way you can write this is 
if (num!=Number(num)){
   return -1
}
return Number(numOfDice)

This way you can eliminate the else statement. This is possible as the if statement already contains a return hence only in scenarios where the if statement condition is not fulfilled will the function return Number(numOfDice). In cases where the if condition is met, JS will exit the function after returning -1 and would not run the second return statement

File: script.js [hunk 2]
```diff
@@ -1,4 +1,208 @@
...
+var autoGeneratedList = function (diceList, mode) {
+  var sortedDiceList;
+  if (mode == "highest") {
+    sortedDiceList = diceList.sort(function (a, b) {
+      return a - b;
+    });
+  } else if (mode == "lowest") {
+    sortedDiceList = diceList.sort(function (a, b) {
+      return b - a;
+    });
```
Comment (emily-zhangg) [inline script.js:54]: good job looking up how to us a sort function

File: script.js [hunk 3]
```diff
@@ -1,4 +1,208 @@
...
+      index = index + 1;
+    }
+  }
+  console.log(
+    "current winner index",
+    currentWinnerIndex,
+    currentWinningScore,
+    mode
+  );
+  return [currentWinnerIndex, currentWinningScore, mode];
```
Comment (emily-zhangg) [inline script.js:113]: great job here! really like the use of an array to return 3 values from a function!!

File: script.js [hunk 4]
```diff
@@ -1,4 +1,208 @@
+//define game mode
+var game_state_select_num_of_players = "select number of players";
+var game_state_waiting_for_username = " waiting for player username";
+var game_state_select_num_of_dice = "player select num of dice";
+var game_state_roll_dice = "player roll your dice";
+var game_state_autogenerate_order = "player computer autogenerates order";
+var game_state_find_winner = "find winner";
+var currentGameState = game_state_select_num_of_players;
```
Comment (emily-zhangg) [inline script.js:8]: nice use of global variables for game mode

**Review by yiqiingg — COMMENTED** [review]
great job valerie!! I like the use of global variables and the separation of logic into helper functions here.

**Pull request closed without merging**